checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -170,7 +170,22 @@ protected boolean isInitializationAnnotation(AnnotationMirror anno) {
      * @param field field that might have invariant annotation
      * @return whether or not field has the invariant annotation
      */
-    protected abstract boolean hasFieldInvariantAnnotation(VariableTree field);
+    protected boolean hasFieldInvariantAnnotation(VariableTree field) {
```

Comment:
At the moment it isn't clear how to choose between these two methods and which one to override.
Adding final will clarify which one to override.
Maybe adding a comment to this method can clarify that it's just a convenience method for the other one.